portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login-status")
+public class LoginStatusServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    PrintWriter out = response.getWriter();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // If user is not logged in, return a message with a link to login
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL("/index.html");
+      String message = "<p>Login <a href=\"" + loginUrl + "\">here</a> to post a comment.</p>";
+      JsonObject json = new JsonObject();
+      json.addProperty("message", message);
+      out.println(json.toString());
+      return;
+    }
+
+    // User should have set a username at this point
+    String username = getUserUsername(userService.getCurrentUser().getUserId());
+    if (username == null) {
+      System.out.println("User did not set a username");
```

Comment:
Is there a way to specify a schema for your data store, that says "username cannot be null"? That might simplify this logic by cutting the problem off at its root.